pkg/kv/kvserver/replica.go

```diff
@@ -442,6 +442,13 @@ type Replica struct {
 	shMu struct {
 		// The state of the Raft state machine.
 		state kvserverpb.ReplicaState
+		// raftTruncState contains the raft log truncation state, i.e. the ID of the
+		// last entry of the log prefix that has been compacted out from the raft
+		// log storage.
+		//
+		// Invariant: raftTruncState == *state.TruncatedState.
```

Comment:
self-nit: this invariant was true in the second commit, but turned into `state.TruncatedState == nil` towards the end. Fix the comment.